### ISSUE TITLE
COVID-19 primary care codes

### ISSUE THREAD

**Issue by kevwing** [opensafely/codelist-development#22]

**Comment by kevwing** [comment]
Latest version of the codelist for review (although this one does not have Laurie's latest comments included), the red ones are the ones that need reviewed.

[COVID_readCodes_Counts_v3.0.xlsx](https://github.com/opensafely/codelist-development/files/4798408/COVID_readCodes_Counts_v3.0.xlsx)

**Comment by krishnanbhaskaran** [comment]
Reproducing the most recent conversation in the lost thread - my comments on v3 were questioning whether the "Recently performed a <medical procedure>"/"Recently worked in a <high risk setting>" codes (rows 28-82) should be downgraded to "unrelated to case status" and similarly whether the various non-specific assessment codes (rows 83-91) e.g. sit-to-stand test) should be downgraded to unrelated.

Laurie's comments were:
Hi Kevin
I have mainly agreed but highlighted a few where I didn't
I see KB's points above but I would imagine that a GP wouldn't be coding these unless a pt was consulting for COVID - (and I doubt they will be used much at all) but good to check.
With the ab +ve people I thought we were coding them as historic case but as discussed we will need to consider the PPV in some way.
With the non-specific assessment codes I think that a GP wouldn't be recording unless they were considering COVID but without a +ve or -ve comment I don't think we can be clearer than possible case

[COVID_readCodes_Counts_v3.0_LT.xlsx](https://github.com/opensafely/codelist-development/files/4798983/COVID_readCodes_Counts_v3.0_LT.xlsx)

**Comment by CarolineMorton** [comment]
I mostly agree with the above comments. 

With Row 30 to 82, they are currently potential cases. If we mean, potential exposed then this is perhaps fine but i think we should be carefully because we are going to get people without symptoms who work in healthcare who are being screened as part of a population monitoring scheme, and I am not sure this is really the same as being a contact of a known case, like a household member. I would suggest we change the name of potential case to exposure to disease or something similar because it is easy to interpret as v similar to suspect case. 

I'm unclear about row 83 - 91 as these seem like clinical findings that could be performed for other things as well as covid although they were massively popularised during the epidemic and the switch to telephone triage. @kevwing are you able to clarify how these came to be on the list? If they were added during the outbreak, i think it is safe to assume they are related to covid assessment in suspected cases; but if they were available prior to this, then we should be more cautious. 

Row 126  - 132 do not have read codes in Column A which will mean that it will cause an error in the backend code as this is the key part that is searched on. 

Agree with  @laurietomlinson  on row 133 - 137. These should be historical cases. 

Row 104 doesn't exist as far as i know yet (!) but I think this should be excluded as it is a procedure rather than a finding or situation.

**Comment by kevwing** [comment]
Thanks to all for review/thread-saving detective work. Version 3.1 attached with I think all changes implemented from comments so far, pending responses to any issues raised below.

[COVID_readCodes_Counts_v3.1.xlsx](https://github.com/opensafely/codelist-development/files/4802843/COVID_readCodes_Counts_v3.1.xlsx)

@CarolineMorton in response to your queries/points:

1. I've added a new case status that is "exposure to disease" and changed rows 28-82 to this status. I think we need to keep the "potential case" status though as there are a number of entries that we assigned to this value that are not related to exposure i.e. 12 - 14, 83 - 87, 90 - 91 (another option that I am in favour of would be to reassign 12 - 14, 83 - 87, 90 - 91  to "suspect case" so that we can remove "potential case" completely as it is a bit vague).

2. rows 83-91, these were on the snomed list release 26th May 2020 with the following comment: "Note: A new set of exertional desaturation tests have been added. These tests are not specific to COVID-19, but may be of value in patient assessment" so I would propose to leave with the current potential/suspect assignment following changes made after Laurie's comments (or just suspect assignment based on point 1. above).

3. Row 126 - 132 (and any others without read codes) were provided by chris bates by slack on June 5th as DRAFT terms, which chris said he would provide codes for but hasn't yet, I have sent a few reminders. I am still struggling to find online a searchable resource that will include all the UP TO DATE COVID-19 terms where I can just type in a term and output a ctv3 read code, if anyone knows of one please provide me with the link so I can stop pestering chris bates all the time (e.g. as a test it needs to be able to find a ctv3 read code for the term "Severe acute respiratory syndrome coronavirus 2 antibody detection result positive (finding)"). 

4. Exclusion of row 104 as it is a procedure - there are quite a few codes that are procedures (e.g. also rows 98, 99, 102), I think based on previous discussions we have assigned these as either suspect case for acute phase testing or potential historic for antibody testing. So to be consistent with these other decisions, I think it should be left as a potential historic case? 

In other news, there is a [new release (10th June)](https://hscic.kahootz.com/connect.ti/COVID19_info_sharing/view?objectId=67227941) of SNOMED codes that I have not yet checked against our list for new entries. Given how long it is taking for us to check new terms against CTV3 codes and to review, I would suggest we just plough on and try and get this list (v3.1) finalised (i.e. resolutions to points 1-4 above), so we can at least work towards starting to get feasibility counts using the codes and case assignments that we have already. Then I can work in parallel to update our list with any new codes, so that it won't hold up getting usable codelists onto https://codelists.opensafely.org/.

If people think further discussion is needed in relation to points 1-4 above, I'd suggest another meeting.

Thanks,

Kevin

**Comment by krishnanbhaskaran** [comment]
I agree with version 3.1 above. I would support Kevin's suggestion of rolling "potential" in with "suspect", as I don't think there is any clear conceptual clear distinction between them. K

**Comment by krishnanbhaskaran** [comment]
p.s. I agree we should finalise and implement this list for now, and then update as necessary with the new release later

**Comment by DarthCTR** [comment]
Looks great, @kevwing . Thanks for putting these codes together. Agree with KB comments.

**Comment by kevwing** [comment]
Hi All, here is the latest version of the COVID codes as a combined file.

Of the codes that Chris Bates provided last week that are used by Pillar 2, the there were 6 (out of 24) that were not included in our list so hadn't been assigned a case status, these are marked in red in the attached, please let me know if anyone disagrees with the case status I've assigned them.

[COVID_readCodes_Counts_v3.2.xlsx](https://github.com/opensafely/codelist-development/files/4811377/COVID_readCodes_Counts_v3.2.xlsx)

Next step I was planning is to split this up into separate files by case status and upload these onto https://codelists.opensafely.org/.

Thanks,

Kevin

**Comment by CarolineMorton** [comment]
Hi Kevin, 

Thanks for this. 
I am not sure about:
- Does not have immunity to SARS-CoV-2 (severe acute respiratory syndrome coronavirus 2) - being a potential historical case as i think this suggests that they have had a negative antibody test. Happy to be persuaded otherwise though if others disagree.

**Comment by kevwing** [comment]
Hi Caroline,

I think our approach for similar terms to this one (e.g. SARS-CoV-2 (severe acute respiratory syndrome coronavirus 2) IgM detection result negative) was to put "potential historic" because of the current uncertainty whether everybody who has had the disease would positive for antibodies, the thinking being that if they have been tested for antibodies, there would likely have been a reason for doing so (making them "potential historic").

**Comment by CarolineMorton** [comment]
Ok I am happy with that. Thanks for answering my query - sorry to rehash things again!!

**Comment by krishnanbhaskaran** [comment]
hi - thanks Kevin agree with the 6 new red ones. I notice that those 6 are all referring to antibody testing, but I thought pillar 2 was focussed on antigen/"current infection" testing - were the other codes in the 24 about current infection?

I agree with next step to get the lists onto opensafely ready for use.

**Comment by kevwing** [comment]
Only 4/24 of the codes provided by Chris as Pillar 2 relate to current infection (see attached). It may be that these are Pillar 2 and Pillar 3 in that case. Will check with Chris.

[Pillar2_TPP_Codes_fromChrisBates.xlsx](https://github.com/opensafely/codelist-development/files/4812902/Pillar2_TPP_Codes_fromChrisBates.xlsx)

**Comment by krishnanbhaskaran** [comment]
from Slack re 111 codes: 

@channel just been informed of a national 111 pathways problem where actual Covid dispositions stopped going out as a Covid dispositions and went out as “home care”. This means there is no corresponding code in GP systems. This started on 11th May and is in the process of being fixed. Don’t know when the fix will be applied to systems. Don’t know if we’ll get fixed data - but sounds almost impossible we think.

Krishnan Bhaskaran  1 hour ago
@chris Related to this, we noticed that none of the codes we currently have on "covid in primary care" specifically mention/refer to 111.  (https://github.com/opensafely/codelist-development/issues/22)
Might we be missing a set of codes that capture 111-related records? If so do you know if the relevant codes have been brought together anywhere?
cc @Kevin @Laurie Tomlinson

Laurie Tomlinson  1 hour ago
I'm also unclear on this - from what I understood yesterday it was channeled into the GP record but not as 111 code specifically but I may have misunderstood

chris  1 hour ago
111 disposition codes don’t end up coded in GP records as a rule. However, because of the situation we wrote code to hijack the Covid dispositions to put an actual CTV3 code in place. It was a piece of work with NHSX and NHSD. You DO have that code in your list. The issue is that some of the Covid 111 calls after 11th May did not come down with the right “hijackable” disposition.
Not sure if EMIS ever did the same btw?

Krishnan Bhaskaran  43 minutes ago
Thanks Chris that's helpful. Do you know which specific code in the list it is that captures this - would it be Y212c "Advice given about severe acute respiratory syndrome coronavirus 2 by telephone (situation)" perhaps? I couldn't find any in the spreadsheet that specifically mention 111 but may be missing it!

Jonny  32 minutes ago
The code we add for incoming 111 covid dispositions is Y20cf - This maps to 1240761000000102 - Suspected COVID-19. This was the code requested by NHSE/X

Jonny  30 minutes ago
At the moment, it is not possible to distinguish coded entries with that code in OpenSAFELY that were added via incoming 111 messages from those added manually via the patient record

Krishnan Bhaskaran  29 minutes ago
Thanks Jonny that makes things much clearer.

Jonny  27 minutes ago
...but we could maybe build something in to distinguish the codes from 111 if necessary

**Comment by kevwing** [comment]
Two new "helper: codelists have been added to the OpenSAFELY codelists that provide further information about some of the codes in the 7 'COVID identifcation in primary care' codelists (and are related to the discussion above), as follows:

1. "COVID Identification in primary care (helper) - 111 suspected" = suspect codes used by 111
2. "COVID Identification in primary care (helper) - Pillar 2 onwards" = case, negative test, potential historic and historic codes used in Pillar 2 testing onwards

**Comment by krishnanbhaskaran** [comment]
[covid primary care codes subdivisions for exploration v1.0.xlsx](https://github.com/opensafely/codelist-development/files/4856975/covid.primary.care.codes.subdivisions.for.exploration.v1.0.xlsx)

Hi all. Further to yesterday's discussions re preliminary outputs, I have proposed subdivions of the "case" and "suspected" codes (I think these are the main categories of interest with the others of more secondary concern). In summary I have broken them down into:


Case:
--
test positive
clinical or nos covid
covid sequelae
 
 
Suspect:
suspected code
tested
isolated
non-specific covid-relevant clinical assessment
advice given

... which I thought would be useful groupings to describe (in terms of numbers of codes, use over time, mortality etc) to help us come to firmer decisions on how to combine for studies.

Please share any thoughts.

**Comment by krishnanbhaskaran** [comment]
Further to above, attached are the more detailed breakdowns of the "case" and "suspect" codes, as presented last week:

CASES:
Positive tests: [exploratory_covid_positive_test.xlsx](https://github.com/opensafely/codelist-development/files/4920057/exploratory_covid_positive_test.xlsx)

Clinical code: [exploratory_covid_clinical_or_nos.xlsx](https://github.com/opensafely/codelist-development/files/4920058/exploratory_covid_clinical_or_nos.xlsx)

Covid sequelae: [exploratory_covid_sequelae.xlsx](https://github.com/opensafely/codelist-development/files/4920059/exploratory_covid_sequelae.xlsx)

SUSPECTED:
Suspected codes: [exploratory_covid_suspected_codes.xlsx](https://github.com/opensafely/codelist-development/files/4920061/exploratory_covid_suspected_codes.xlsx)

Had test: [exploratory_covid_had_test.xlsx](https://github.com/opensafely/codelist-development/files/4920062/exploratory_covid_had_test.xlsx)

Non-specific clinical assessments: [exploratory_covid_nonspec_clinical_assessment.xlsx](https://github.com/opensafely/codelist-development/files/4920067/exploratory_covid_nonspec_clinical_assessment.xlsx)

Isolated: [exploratory_covid_isolated.xlsx](https://github.com/opensafely/codelist-development/files/4920064/exploratory_covid_isolated.xlsx)

Given advice: [exploratory_covid_advice.xlsx](https://github.com/opensafely/codelist-development/files/4920065/exploratory_covid_advice.xlsx)

The exploratory analysis of these is at: https://github.com/opensafely/covid-positive-infection-research/tree/master/released_analysis_results

If useful for studies, these can be put onto codelists.opensafely - would need a clinical sign off (@laurietomlinson or @CarolineMorton?)

EPI SIGN OFF: Krishnan Bhaskaran 14/7/2020
CLINICAL SIGN OFF:
